tasks/src/main/java/io/MainTask.java

```diff
@@ -0,0 +1,106 @@
+package io;
+
+import static java.nio.file.FileVisitResult.CONTINUE;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+class FileTreeVisitor extends SimpleFileVisitor<Path> {
+
+  static String fileVisitResult = "";
+
+  @Override
+  public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
+    fileVisitResult = fileVisitResult + "  |---" + dir.getFileName() + "\n";
+    return CONTINUE;
+  }
+
+  @Override
+  public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
+    fileVisitResult = fileVisitResult + "  |  " + file.getFileName() + "\n";
+    return CONTINUE;
+  }
+}
+
+public class MainTask {
+
+  private static void getInfoAboutFile(Path path) throws IOException {
+    File file = path.toFile();
+
+    int countFiles = 0;
+    int countFolders = -1; //not count the main folder
+    int nameFiles;
+    int lengthNameFiles = 0;
+    List<String> listOfLines = new ArrayList<>();
+    List<String> listFiles = new ArrayList<>();
+
+    try (FileReader reader = new FileReader(file)) {
+      Scanner sc = new Scanner(reader);
+      while (sc.hasNext()) {
+        String lineNext = sc.nextLine();
+        listOfLines.add(lineNext.replace("|", " ").replace("-", " ").trim());
+        listOfLines.removeIf(line -> (line.length() == 0)); //remove empty lines
+      }
+
+      if (!listOfLines.isEmpty()) {
+        for (String lines : listOfLines) {
+          if (lines.matches(".+(\\.[a-z0-9]{2,4})$")) {  //find all expansion
+            countFiles++;
+            listFiles.add(lines); //if the file name matches regex then add files to the list
```

Comment:
Your comments make sense, but you must write your code in a way that another programmer could understand what is going on in each line without comments. Instead of writing comments on tricky code snippets I'd suggest you split complicated code to small separate auxiliary methods with good names that will give a clear understanding of the purpose that method was implemented for.
Example:
```
getInfoAboutFile() {
var1;
var2;
var3;
getListOfLines(var1);
getListOfFiles(var2);
calculateAverageQtyOfFilesInFolder(var3);
calculateAverageLengthOfFileNames(var3);
}
```